tests/test_review_inference_preflight_contract.py

```
@@ -0,0 +1,260 @@
+    token_file.chmod(0o600)
+    curl_double = tmp_path / "curl"
+    curl_double.write_text(
+        '#!/bin/sh\ncase "$*" in *unit-secret-token*) exit 1;; esac\n[ -z "${INFERENCE_TOKEN+x}" ] || exit 2\nIFS= read -r header\n[ "$header" = \'Authorization: Bearer unit-secret-token\' ] || exit 3\n'
+    )
+    curl_double.chmod(0o700)
+    environment = {
+        key: value for key, value in os.environ.items() if key != "INFERENCE_TOKEN"
+    }
+    environment["PATH"] = str(tmp_path) + os.pathsep + environment.get("PATH", "")
+    subprocess.run(
+        ["sh", "-c", example],
```

Ruff (0.16.3)

[error] Starting a process with a partial executable path

(S607)